### PR TITLE
Usage of mage secrets as a secondary to when there are no aws key or …

### PR DESCRIPTION
**Pull request by jamesking-github**
…aws secret in the environment variables

# Description
AWS access keys and access secrets are currently required to be environment variables, mage secrets are better suited.  This change been designed to keep the environment variable being the primary source of AWS Credentials with Mage secrets only used if the environment variables are not set.

# How Has This Been Tested?
In Progress currently

# Checklist
- [x] The PR is tagged with proper labels (bug, enhancement, feature, documentation)
- [x] I have performed a self-review of my own code
- [ ] I have added unit tests that prove my fix is effective or that my feature works
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation

cc:

### CODE REVIEW

**Comment by wangxiaoyou1993** [comment]
@dy46 could you help review this PR?

**Comment by dy46** [comment]
@jamesking-github is there a reason this is a draft?

**Comment by jamesking-github** [comment]
Hey. I moved it to draft for two reasons:I didn’t manage to positively test it locally as the instructions to create a development environment under contribute didn’t work for me AndI thought the eks and ecs creds getting should also get these improvements assuming I can prove they work. Let me know if you have other ideas as very happy to discuss them. Thanks as ever David. On 28 Sep 2023, at 21:31, David Yang ***@***.***> wrote:﻿
@jamesking-github is there a reason this is a draft?

**Comment by dy46** [comment]
got it, that makes sense to me.

**Comment by jamesking-github** [comment]
@dy46 I have included the updates to the other AWS services and updated the documentation.  Issues getting Mage to run from my branch has limited the testing I can do however I think there is minimal chance of regression as a result of these changes so I have moved it back to open

File: mage_ai/services/aws/__init__.py [hunk 1]
```diff
@@ -1,8 +1,30 @@
...
 
+from mage_ai.data_preparation.shared.secrets import get_secret_value
+
 
 def get_aws_region_name():
-    return os.getenv(
-        'AWS_REGION_NAME',
-        os.getenv('AWS_DEFAULT_REGION', 'us-west-2'),
-    )
+    region_name = os.getenv('AWS_REGION_NAME')
```
[the remark this answers is not in the document]
Comment (jamesking-github) [reply mage_ai/services/aws/__init__.py:7]: Project or file level?

[the remark this answers is not in the document]
Comment (jamesking-github) [reply mage_ai/services/aws/__init__.py]: Module level dunder constants created

[the remark this answers is not in the document]
Comment (dy46) [reply mage_ai/services/aws/__init__.py]: @jamesking-github we use all caps for constant names in mage. for example, `AWS_REGION_NAME`

[the remark this answers is not in the document]
Comment (jamesking-github) [reply mage_ai/services/aws/__init__.py]: Updated as requested

File: mage_ai/services/aws/__init__.py [hunk 2]
```diff
@@ -2,34 +2,34 @@
...
 from mage_ai.data_preparation.shared.secrets import get_secret_value
 
-__aws_region_name__ = "AWS_REGION_NAME"
-__aws_default_region__ = "AWS_DEFAULT_REGION"
-__aws_access_key_id__ = "AWS_ACCESS_KEY_ID"
-__aws_secret_access_key__ = "AWS_SECRET_ACCESS_KEY"
+AWS_REGION_NAME = "AWS_REGION_NAME"
+AWS_DEFAULT_REGION = "AWS_DEFAULT_REGION"
+AWS_ACCESS_KEY_ID = "AWS_ACCESS_KEY_ID"
+AWS_SECRET_ACCESS_KEY = "AWS_SECRET_ACCESS_KEY"
```
Comment (dy46) [inline mage_ai/services/aws/__init__.py:8]: we use single quotes for python strings in mage

File: mage_ai/services/aws/ecs/ecs.py [hunk 3]
```diff
@@ -64,8 +68,13 @@ def stop_task(task_arn: str, cluster: str = None) -> None:
...
 def list_tasks(cluster) -> List[Dict]:
     region_name = get_aws_region_name()
+    aws_access_key = get_aws_access_key_id()
+    aws_secret_key = get_aws_secret_access_key()
     config = Config(region_name=region_name)
-    ecs_client = boto3.client('ecs', config=config)
+    ecs_client = boto3.client('ecs',
+                              aws_access_key_id=aws_access_key,
+                              aws_secret_access_key=aws_secret_key,
+                              config=config)
```
[the remark this answers is not in the document]
Comment (jamesking-github) [reply mage_ai/services/aws/ecs/ecs.py]: Done

**Review by dy46 — APPROVED** [review]

**Pull request merged**